### PR TITLE
4D mask documentation updates

### PR DESCRIPTION
**Pull request by poedator**
following https://github.com/huggingface/transformers/pull/27539 this PR adds updates to transformers documentation to reflect possibility of utilizing 4D masks.

Plan:
- add updates for Llama model docstring(s)
- identify other models that can use 4D masks in present form (which requires ability to accept custom `position_ids` argument) and updating their docstrings. Classes that need updates:
    -  Falcon Model
    - [TODO identify more]
- update code comments that may need corrections, like cases where the mask may be either 2D or 4D now. one example is based on [this comment](https://github.com/huggingface/transformers/pull/27539#issuecomment-1863285474) by @shentianxiao

Update 20.12.2023:
to find out which models require docstring changes, I scanned all model classes in transformers insing inspect.
- excluded tf and jax classes
- excluded models without `position_ids` argument in `.forward()` - can't use 4D mask effectively
- excluded models that do not use `_prepare_4d_attention_mask` method - need different code change to use 4D mask
- excluded multi-modal models (clip, clvp, vit, bark, git)

what is left is LlamaModel, FalconModel and XGLMModel.

cc @ArthurZucker

### CODE REVIEW

**Comment by ArthurZucker** [comment]
Feel free to ping me for a review whenever this is ready 🤗

**Comment by poedator** [comment]
> Feel free to ping me for a review whenever this is ready 🤗

@ArthurZucker , I only identified 3 applicable model classes and made changes. Please check my logic in classes selection in my big first message above.

**Review by ArthurZucker — COMMENTED** [review]
#28132 might be relevant to you! I'll make sure 4d can still be used but will make things a lot easier I think.
LGTM otherwise!

**Comment by alex-hh** [comment]
Has something like this or equivalent been merged? I don't see much documentation for 4D masks but would find useful!

**Comment by ArthurZucker** [comment]
Yep sorry, it's supported for some models. https://github.com/huggingface/transformers/blob/18527bd029316037dd174c85b1f9b55288b50c6c/src/transformers/models/bloom/modeling_bloom.py#L65 doc is a bit scarce, feel free to open a pr to add this!
